File: Projects/MultiFileReader/modules/csv_parser.py
#
# Nimish Nayak
# 10-06-2017
#

#
# CSV Parser 
#

# required libraries
import logging
import csv
from parser import Parser
from utils import Utils

class CsvParser(Parser):

    # initialize the instance variables
    def __init__(self, input_file):
        # call the base class constructor
        Parser.__init__(self, input_file)
        logging.debug("Csv parser initialized")

    # parse the file using a csv parser 
    def parse_file(self):
        # get a list of CSV tags from the document and create the list of dictionaries 
        with open(self.input_file, "rb") as f:
            users = [row for row in csv.reader(f)][1:]    
        logging.debug("%d users:" % len(users))
        for user in users:
            
            # default values 
            val = {"name":"","active":False,"value":0}
            
            # check if csv 
            if len(user) == 3:
                if user[0]:
                    val["name"] = user[0]
                if user[1]:
                    val["active"] = Utils.str_to_bool(user[1])
                if user[2]:
                    val["value"] = int(user[2])
            self.data.append(val)
        logging.debug("data saved: %s"%self.data)

File: Projects/MultiFileReader/modules/csv_parser.php
<?php

#
# Nimish Nayak
# 10-06-2017
#

#
# CSV Parser 
#

require_once("./modules/utils.php");
require_once("./modules/parser.php");

class CsvParser extends Parser {

    # initialize the instance variables
    function __construct($inputFile){
        
        # call the base class constructor
        parent::__construct($inputFile);
        echo("Csv parser initialized\n");
    }
    
    # parse the file using a csv parser 
    public function parseFile(){

        $fp = fopen($this->inputFile,"r");

        $this->data = array();

        while (!feof($fp)) {

            $lineArray = fgetcsv($fp);

            # create an associative array from the line 
            # name,active,value
            # John,true,500

            $dict = array("name" => $lineArray[0], "active" => Utils::strToBool($lineArray[1]), "value" => (int)$lineArray[2]);
            array_push($this->data, $dict);

        }

        fclose($fp);
        # var_dump($this->data);
    }

}

?>